code/Language/Drasil/Reference.hs

```diff
@@ -88,11 +93,25 @@ changeMap cs = Map.fromList $ zip (map (^. uid) (lcs ++ ulcs))
 bibMap :: [Citation] -> BibMap
 bibMap cs = Map.fromList $ zip (map (^. uid) scs) (zip scs [1..])
   where scs :: [Citation]
-        scs = sortBy citeSort cs
+        scs = sortBy uidSort cs
         -- Sorting is necessary if using elems to pull all the citations
         -- (as it sorts them and would change the order).
         -- We can always change the sorting to whatever makes most sense
 
+conGrp :: ConceptChunk -> ConceptChunk -> Bool
+conGrp a b = (cdl a) == (cdl b) where
+  cdl :: ConceptChunk -> UID
+  cdl x = if length (x ^. cdom) == 1 then head $ x ^. cdom
```

Comment:
```haskell
cdl x = the_dom $ x ^. cdom
  where
    the_dom [d] = d
    the_dom ds = error $ "Expected ConceptDomain for: " ++ (x ^. uid) ++
           " to have a single domain, found " ++ (show $ length ds) ++
           " instead."
```
Basically `head` should be avoided at all cost (same with other partial operations), even when guarded as you did. It is better to check for length 1 structurally.